tiledb/sm/filter/checksum_md5_filter.cc

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file   checksum_md5_filter.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2020 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class ChecksumMD5Filter.
+ */
+
+#include "tiledb/sm/filter/checksum_md5_filter.h"
+#include "tiledb/sm/buffer/preallocated_buffer.h"
+#include "tiledb/sm/encryption/encryption.h"
+#include "tiledb/sm/encryption/encryption_key.h"
+#include "tiledb/sm/enums/encryption_type.h"
+#include "tiledb/sm/enums/filter_type.h"
+#include "tiledb/sm/misc/logger.h"
+#include "tiledb/sm/tile/tile.h"
+
+namespace tiledb {
+namespace sm {
+
+ChecksumMD5Filter::ChecksumMD5Filter()
+    : Filter(FilterType::FILTER_CHECKSUM_MD5) {
+}
+
+ChecksumMD5Filter* ChecksumMD5Filter::clone_impl() const {
+  auto clone = new ChecksumMD5Filter;
+  // Copy key bytes buffer.
```

Comment:
Is this just because this class has no private members? If so, I would remove the current comment and add a new one.